### PR TITLE
Use HttpJsonRequest instead of HttpJsonHelper

### PR DESCRIPTION
**Pull request by garagatyi**
@skabashnyuk @sleshchenko @evoevodin Please review

### CODE REVIEW

**Comment by skabashnyuk** [comment]
ok

**Comment by skabashnyuk** [comment]
Can we remove HttpJsonHelper after that?

**Comment by garagatyi** [comment]
Yes, we can!

File: core/che-core-api-core/src/main/java/org/eclipse/che/api/core/rest/RemoteServiceDescriptor.java [hunk 1]
```diff
@@ -95,8 +97,11 @@ public ServiceDescriptor getServiceDescriptor() throws IOException, ServerExcept
                 myServiceDescriptor = serviceDescriptor;
                 if (myServiceDescriptor == null) {
                     try {
-                        myServiceDescriptor = serviceDescriptor = HttpJsonHelper.options(getServiceDescriptorClass(), baseUrl);
-                    } catch (NotFoundException | ConflictException | UnauthorizedException | ForbiddenException e) {
+                        myServiceDescriptor = serviceDescriptor = requestFactory.fromUrl(baseUrl)
```
Comment (sleshchenko) [inline core/che-core-api-core/src/main/java/org/eclipse/che/api/core/rest/RemoteServiceDescriptor.java]: WDYT about rewriting this method without myServiceDescriptor variable?

```
        if (serviceDescriptor == null) {
            synchronized (this) {
                if (serviceDescriptor == null) {
                    try {
                        return serviceDescriptor = requestFactory.fromUrl(baseUrl)
                                                          .useOptionsMethod()
                                                          .request()
                                                          .as(getServiceDescriptorClass(), null);
                    } catch (NotFoundException | ConflictException | UnauthorizedException | ForbiddenException e) {
                        throw new ServerException(e.getServiceError());
                    }
                }
            }
        }
```

**Comment by sleshchenko** [comment]
LGTM

**Comment by voievodin** [comment]
ok

File: core/che-core-api-core/src/main/java/org/eclipse/che/api/core/rest/RemoteServiceDescriptor.java (hunk 1 above)
Comment (garagatyi) [inline core/che-core-api-core/src/main/java/org/eclipse/che/api/core/rest/RemoteServiceDescriptor.java]: Looks OK for me from concurrency POW. @evoevodin WDYT?

Comment (voievodin) [inline core/che-core-api-core/src/main/java/org/eclipse/che/api/core/rest/RemoteServiceDescriptor.java]: +1

**Comment by tareksha** [comment]
Hi,
can this be down-ported to 3.x ?
@garagatyi @skabashnyuk @sleshchenko @evoevodin

**Comment by skabashnyuk** [comment]
@tareqhs Yes. If you want you can do so.

**Pull request merged**